packages/compass-assistant/src/assistant-provider.tsx

```diff
@@ -0,0 +1,60 @@
+import { DrawerSection } from '@mongodb-js/compass-components';
+import React, { type PropsWithChildren, useCallback, useRef } from 'react';
+import { type UIMessage, useChat } from './@ai-sdk/react/use-chat';
+import type { Chat } from './@ai-sdk/react/chat-react';
+import { AssistantChat } from './assistant-chat';
+import { usePreference } from 'compass-preferences-model/provider';
+
+export const ASSISTANT_DRAWER_ID = 'compass-assistant-drawer';
+
+import { createContext, useContext } from 'react';
+
+type AssistantActions = unknown;
+
+export const AssistantActionsContext = createContext<AssistantActions>({});
+
+export function useAssistantActions(): AssistantActions {
+  return useContext(AssistantActionsContext);
+}
+
+export const AssistantProvider: React.FunctionComponent<
+  PropsWithChildren<{
+    chat: Chat<UIMessage>;
+  }>
+> = ({ chat, children }) => {
+  const enableAIAssistant = usePreference('enableAIAssistant');
+
+  const { messages, sendMessage } = useChat({
+    chat,
+  });
+
+  const contextActions = useRef({});
+
+  const handleMessageSend = useCallback(
+    (messageBody: string) => {
+      void sendMessage({ text: messageBody });
+    },
+    [sendMessage]
+  );
+
+  if (!enableAIAssistant) {
+    return <>{children}</>;
+  }
+
+  return (
+    <AssistantActionsContext.Provider value={contextActions.current}>
+      <DrawerSection
```

Comment:
This is what I was talking about in slack: Just move this <DrawerSection> somewhere lower down in the render tree, keeping <AssistantActionsContext.Provider where it is.

Reply:
(Obviously not necessary for this PR, though)